application/src/main/java/org/togetherjava/tjbot/features/mediaonly/MediaOnlyChannelListener.java

```diff
@@ -45,7 +45,15 @@ public void onMessageReceived(MessageReceivedEvent event) {
         }
 
         if (messageHasNoMediaAttached(message)) {
-            message.delete().flatMap(any -> dmUser(message)).queue();
+            message.delete()
+                .flatMap(any -> message.getAuthor()
+                    .openPrivateChannel()
+                    .flatMap(channel -> channel.sendMessage(warningMessage(message))))
+                .queue(any -> {
+                }, failure -> message.getChannel()
+                    .sendMessage(warningMessage(message))
+                    .queue(notificationMessage -> notificationMessage.delete()
+                        .queueAfter(1, TimeUnit.MINUTES)));
```

Comment:
this is really long and complex now. not sure why u removed `dmUser`. please change this code by using utility methods so that in the end its just:
```java
message.delete()
  .flatMap(any -> dmUser(message))
  .queue(any -> {}, failure -> tempNotifyUserInChannel(message));
```

Reply:
https://github.com/Together-Java/TJ-Bot/pull/769#discussion_r1103854200

I was asked it to modify it the other way

Reply:
@Zabuzard whats the problem with current method? a single method to get the message and send it where wanted

Reply:
@Taz03 
Is this a better approach ??
(or )
shall I make 3 functions 
1. message Building
2. Dm user
3. Temporary message?

Reply:
isn't that what zabu said as well? im ok with it

Reply:
which one 3 functions @Taz03

Reply:
I did resolve it this time check it out